src/Parallel/Pin.cpp

```diff
@@ -55,6 +57,52 @@ seissol::parallel::Pinning::Pinning() {
   openmpMask = getWorkerUnionMask();
 }
 
+void seissol::parallel::Pinning::checkEnvVariables() {
+  const auto rank = MPI::mpi.rank();
+  if (const char* envVariable = std::getenv("SEISSOL_FREE_CPUS_MASK")) {
+    auto parsedResult = seissol::IntegerMaskParser::parse(std::string(envVariable));
+    if (parsedResult) {
+      parsedFreeCPUsMask = parsedResult.value();
+
+      bool isMaskGood{true};
+      const auto numLocalProcesses = MPI::mpi.sharedMemMpiSize();
+      if (numLocalProcesses > parsedFreeCPUsMask.size()) {
+        logInfo(rank) << "Num. local processes is greater than free cpus mask";
+        isMaskGood = false;
+      }
```

Comment:
Must be an == here. Otherwise the code will break later if numLocalProcesses < parsedFreeCPUsMask.size()

Reply:
Hi, this part of the code checks whether a number of local processes is greater than the provided mask. If it is true it sets `isMaskGood` to `false`.

https://github.com/SeisSol/SeisSol/blob/b550796b6555d3e77b3dc5b1f2e5934d98251b19/src/Parallel/Pin.cpp#L90-L98

If the mask is bad (for some reason) the mask will become empty (see Line 97).

Line `133` is supposed to ensure that we are going to operate on a mask which does not have side-effects e.g., the local MPI rank is greater or equal to the mask size 

https://github.com/SeisSol/SeisSol/blob/b550796b6555d3e77b3dc5b1f2e5934d98251b19/src/Parallel/Pin.cpp#L133-L139

Reply:
Ah, yes, sorry read the if the wrong way... nevermind then.